src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs

```diff
@@ -206,7 +206,7 @@ public int Millisecond
         /// Unix format string
         /// </summary>
         [Parameter(ParameterSetName = "net")]
-        [ValidateSetAttribute("FileDate", "FileDateUniversal", "FileDateTime", "FileDateTimeUniversal", AllowAll=true)]
+        [ArgumentCompleter("FileDate", "FileDateUniversal", "FileDateTime", "FileDateTimeUniversal")]
```

Comment:
This is much better, but I think still a little weird, I still think it's worth considering adding a new attribute `ArgumentCompletions`. It seems like you could take your class `StringArrayArgumentCompleter`, rename it, make it an attribute, and either duplicate a little bit of logic around `ValidateSet` or `ArgumentCompleter` in the tab completion code.

Reply:
Is your only thought that ArgumentCompleter is a strange name and you want to rename it?

Reply:
I wouldn't rename `ArgumentCompleter` even if it is a strange name (which I suppose it is.)

It's a really minor thing, but to me, `ArgumentCompleter` implies some dynamic code to compute the completions, whereas `ArgumentCompletions` is static.

Reply:
I have missed that for a long time - the ability to easily complete a set without limiting the parameter to the completions. 

Is it a reasonable compromise that you have to go IArgumentCompleter if you want to provide rich `CompletionResult`s, or should be enable this with `ArgumentCompletions` too?

Reply:
Done next iteration.